client/src/onload/publicPath.js

```diff
@@ -5,6 +5,10 @@
  * dynamic bundle imports load, otherwise they will fail.
  */
 
+// TODO: See if this is neccessary. I'm not convinced we need dynamic loads at all, but if we do, I
```

Comment:
@Nerdinacan and I talked about this a bit last night, but for anyone else catching up on this -- we do need this here to support serving Galaxy at a proxy prefix that isn't known at runtime.

Pushing it exclusively back into the webpackConfig (which is how it used to work) doesn't solve the problem sufficiently because that route requires a client rebuild at deploy time -- an example of this is the anvil use case where we have a container that has a prebuilt client that gets served to the end user with a prefix.

Reply:
But the reason that we need it is only because you want to do separate dynamic loads. Is that correct?

Reply:
I just did a search for "import(" in the code base and here are the following places we do dynamic loads. 

ListCollectionCreatorModal.js
PairedCollectionCreatorModal.js
PairedListCollectionCreatorModal.js
RuleBasedCollectionCreatorModal.js
WorkflowInvocationStep.vue
Workflow/Editor/modules/layout

The only time this looks even vaguely appropriate is on the Workflow editor because it loads a large package (ELM) for layout of the workflow graph. But even THERE, that package exposes a standard module import scheme.

Why load these things in a separate chunk? What's the upside other than secret build complexity problems?

Reply:
(I can sideline this code change into another PR too, it's straying away from "getting the docs to work".)

Reply:
The size of the galaxy dependency bundle is absolutely gigantic, and we need to be working towards more aggressive splitting, not away from it.  I think the only two of these I am directly responsible for and familiar with adding are the rulebuilder and the workflow editor, which are both absolutely gigantic chunks (multiple megabytes) that we load on demand instead of in the primary modules.

Native modules will be amazing at some point, but that's probably a whole separate unrelated conversation where we can talk about potentially dumping webpack completely.

I guess my question is -- dynamically loaded bundles have been working great and significantly reduce client load time, other than this one minor hiccup in the styleguide.  Is there a compelling reason for *not* using them for now?

Reply:
I can't think of any other complications yet, other than this hiccup, I just saw it and wondered if you were in fact leaning away from the webpack splitting mechanism.

If so.... I will work on webpack to better enable your scheme. :)

Reply:
The redundant entrypoints do duplicate a bunch of code, but these dynamic bundles should not.  That's the whole point.  I don't think I'd call it 'my scheme', though -- it's right there in the webpack docs, including setting the publicpath dynamically at runtime for situations just like this :)

Reply:
My concern is certainly not that it is an evil scheme, but that every time we use a new plugin, package, whatever, it will expect the internal pathInfo to jive with the configs.

Reply:
> The redundant entrypoints do duplicate a bunch of code, but these dynamic bundles should not.

That's interesting! Maybe we should rewrite the entry files to do this then.

Reply:
@Nerdinacan That is the whole point.  We'll have a single entrypoint that dynamically loads all this crap using this mechanism :)

Reply:
Yeah but even now, we can take those 4 entry points and write them with a promise-based dynamic import and immediately get a payoff.

Reply:
@Nerdinacan Yep!